eos-paygd/main.c

```diff
@@ -549,9 +551,12 @@ main (int   argc,
 
   allow_writing_to_boot_partition (FALSE);
 
+  /* If payg_sync_and_poweroff is scheduled, spin the mainloop until it runs */
+  while (timeout_id)
```

Comment:
This is an infinite loop. `timeout_id` is local to this function, nothing will clear it even after the timeout or idle fires.

Reply:
Nothing will clear it, but it is only infinite until `payg_sync_and_poweroff` is executed since at this commit it will call `exit`. I should have made it clear in the comment above, but as you saw it changes later, so it shouldn't matter too much.

The reason I have split the changes that follow as a separate commit was because I wasn't too happy with them, and expected them to change or be dropped during CR.